test/controllers/api/internship_offers/index_test.rb

```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module Api
+  class IndexTest < ActionDispatch::IntegrationTest
+    include ::ApiTestHelpers
+
+    test 'GET #index without token renders :unauthorized payload' do
+      get api_internship_offers_path(params: {})
+      documents_as(endpoint: :'internship_offers/index', state: :unauthorized) do
+        assert_response :unauthorized
+        assert_equal 'UNAUTHORIZED', json_code
+        assert_equal 'wrong api token', json_error
+      end
+    end
+
+    test 'GET #index without api_full_access renders :unauthorized payload' do
+      user = create(:user_operator)
+      operator = user.operator.update(api_full_access: false)
+
+      documents_as(endpoint: :'internship_offers/index', state: :unauthorized) do
+        get api_internship_offers_path(
+          params: {
+            token: "Bearer #{user.api_token}"
+          }
+        )
+
+        assert_response :unauthorized
+        assert_equal 'UNAUTHORIZED', json_code
+        assert_equal 'access denied', json_error
+      end
+    end
+
+    test 'GET #index without params returns all internship_offers available' do
+      user = create(:user_operator)
+      offer_1 = create(:weekly_internship_offer)
+      offer_2 = create(:weekly_internship_offer)
+      offer_3 = create(:weekly_internship_offer, :unpublished)
+
+      documents_as(endpoint: :'internship_offers/index', state: :success) do
+        get api_internship_offers_path(
+          params: {
+            token: "Bearer #{user.api_token}"
+          }
+        )
+
+        assert_response :success
+        assert_equal 2, json_response['internshipOffers'].count
+        assert_equal 2, json_response['pagination']['totalInternshipOffers']
+        assert_equal 1, json_response['pagination']['totalPages']
+        assert_equal true, json_response['pagination']['isFirstPage']
+        assert_equal offer_1.id, json_response['internshipOffers'][1]['id']
+        assert_equal offer_2.id, json_response['internshipOffers'][0]['id']
+      end
+    end
+
+    test 'GET #index with page params returns the page results' do
+      user = create(:user_operator)
+      (InternshipOffer::PAGE_SIZE + 1).times { create(:weekly_internship_offer) }
+
+      documents_as(endpoint: :'internship_offers/index', state: :success) do
+        get api_internship_offers_path(
+          params: {
+            token: "Bearer #{user.api_token}",
+            page: 2
+          }
+        )
+
+        assert_response :success
+        assert_equal 1, json_response['internshipOffers'].count
+        assert_equal InternshipOffer::PAGE_SIZE + 1, json_response['pagination']['totalInternshipOffers']
+        assert_equal 2, json_response['pagination']['totalPages']
+      end
+    end
+
+    test 'GET #index with big page params returns no results' do
```

Comment:
... returns page size instead of results
et du coup le type du résultat devient non homogène...on pourrait ajouter un champ résultat à vide ? II y est déjà ?

Reply:
Oui les résultats sont homogènes, ça retourne un array vide.
J'ai changé le naming du test qui n'était pas clair 👍